CheckDrive.Api/CheckDrive.Application/Services/Authorization/AuthService.cs

```diff
@@ -0,0 +1,90 @@
+﻿using AutoMapper;
+using CheckDrive.Application.Constants;
+using CheckDrive.Application.DTOs.Account;
+using CheckDrive.Application.DTOs.Identity;
+using CheckDrive.Application.Interfaces;
+using CheckDrive.Application.Interfaces.Authorization;
+using CheckDrive.Domain.Enums;
+using CheckDrive.Domain.Exceptions;
+using Microsoft.AspNetCore.Identity;
+
+namespace CheckDrive.Application.Services.Authorization;
+
+public class AuthService : IAuthService
+{
+    private readonly JwtHandler _jwtHandler;
+    private readonly UserManager<IdentityUser> _userManager;
+    private readonly RoleManager<IdentityRole> _roleManager;
+    private readonly IAccountService _accountService;
+    private readonly IMapper _mapper;
+
+    public AuthService(
+        IMapper mapper,
+        JwtHandler jwtHandler,
+        UserManager<IdentityUser> userManager,
+        RoleManager<IdentityRole> roleManager,
+        IAccountService accountService
+    )
+    {
+        _jwtHandler = jwtHandler ?? throw new ArgumentNullException(nameof(jwtHandler));
+        _userManager = userManager ?? throw new ArgumentNullException(nameof(userManager));
+        _roleManager = roleManager ?? throw new ArgumentNullException(nameof(roleManager));
+        _accountService = accountService ?? throw new ArgumentNullException(nameof(roleManager));
+        _mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+    }
+
+    public async Task<string> LoginAsync(LoginDto loginDto)
+    {
+        var user = await _userManager.FindByNameAsync(loginDto.UserName)
+            ?? throw new InvalidLoginAttemptException("Invalid email or password");
+
+        if (user == null || !await _userManager.CheckPasswordAsync(user, loginDto.Password))
```

Comment:
can first check `user == null` be ever true? Also, why not use **is** instead of **==**